platform-modules/batch-models/src/main/scala/org/ekstep/analytics/updater/UpdateDeviceProfileDB.scala

```diff
@@ -1,57 +1,90 @@
 import org.apache.spark.{HashPartitioner, SparkContext}
-import org.apache.spark.rdd.RDD
 import org.ekstep.analytics.framework._
-import org.ekstep.analytics.framework.util.{CommonUtil, JSONUtils}
+import org.ekstep.analytics.framework.conf.AppConf
+import org.ekstep.analytics.framework.util.CommonUtil
 import org.ekstep.analytics.util.Constants
 
 import scala.collection.mutable.Buffer
+
 case class DeviceProfileKey(device_id: String)
 case class DeviceProfileInput(index: DeviceProfileKey, currentData: Buffer[DerivedEvent], previousData: Option[DeviceProfileOutput]) extends AlgoInput
-case class DeviceProfileOutput(device_id: String, first_access: Option[Long], last_access: Option[Long], total_ts: Option[Double], total_launches: Option[Long], avg_ts: Option[Double], device_spec: Option[Map[String, AnyRef]], uaspec: Option[Map[String, String]], state: Option[String], city: Option[String], country: Option[String], country_code:Option[String], state_code:Option[String], state_custom: Option[String], state_code_custom: Option[String], district_custom: Option[String], fcm_token: Option[String], producer_id: Option[String], user_declared_state: Option[String], user_declared_district: Option[String], api_last_updated_on:Option[Long], updated_date: Option[Long] = Option(System.currentTimeMillis())) extends AlgoOutput
 
 object UpdateDeviceProfileDB extends IBatchModelTemplate[DerivedEvent, DeviceProfileInput, DeviceProfileOutput, Empty] with Serializable {
 
-    val className = "org.ekstep.analytics.model.UpdateDeviceProfileDB"
-    override def name: String = "UpdateDeviceProfileDB"
-
-    override def preProcess(data: RDD[DerivedEvent], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[DeviceProfileInput] = {
-        val filteredEvents = DataFilter.filter(data, Filter("eid", "EQ", Option("ME_DEVICE_SUMMARY")))
-        val newGroupedEvents = filteredEvents.map { event =>
-            (DeviceProfileKey(event.dimensions.did.get), Buffer(event))
-        }.partitionBy(new HashPartitioner(JobContext.parallelization)).reduceByKey((a, b) => a ++ b);
-        val prevDeviceProfile = newGroupedEvents.map(f => f._1).joinWithCassandraTable[DeviceProfileOutput](Constants.DEVICE_KEY_SPACE_NAME, Constants.DEVICE_PROFILE_TABLE)
-        val deviceData = newGroupedEvents.leftOuterJoin(prevDeviceProfile)
-        deviceData.map { x => DeviceProfileInput(x._1, x._2._1, x._2._2) }
-    }
+  val className = "org.ekstep.analytics.model.UpdateDeviceProfileDB"
+
+  override def name: String = "UpdateDeviceProfileDB"
+
+  val db = AppConf.getConfig("postgres.db")
+  val url = AppConf.getConfig("postgres.url") + s"$db"
+  val user = AppConf.getConfig("postgres.user")
+  val pass = AppConf.getConfig("postgres.pass")
+
+  val connProperties = new Properties()
+  connProperties.setProperty("Driver", "org.postgresql.Driver")
+  connProperties.setProperty("user", user)
+  connProperties.setProperty("password", pass)
```

Comment:
Move this to a common util code as this is duplicated across two data products. may be `CommonUtil.getPostgresConnectionProps()`